pages/en/lb3/Controlling-data-access.md

```diff
@@ -355,3 +355,165 @@ To upgrade an application from an earlier version of LoopBack 3.x, follow these
 Specify a `DEBUG` environment variable with value `loopback:security:*` for the console to log the lookups and checks the server
 makes as requests come in, useful to understand things from its perspective.
 Do this in your test environment as there may be quite a lot of output.
+
+## Comprehensive AccessType, Property and End-point
+<table>
+  <tbody>
+    <tr>
+      <th>Access Type</th>
+      <th>Property</th>
+      <th>Verb</th>
+      <th>End point</th>
+    </tr>
+    <tr>
+      <td>READ</td>
+      <td>find</td>
+      <td>GET</td>
+      <td>/model-name-plural</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>findById</td>
+      <td>GET</td>
+      <td>/model-name-plural/{id}</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>exists</td>
+      <td>HEAD</td>
+      <td>/model-name-plural/{id}</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>exists</td>
+      <td>READ</td>
+      <td>/model-name-plural/{id}/exists</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>__get__relationName</td>
+      <td>GET</td>
+      <td>/model-name-plural/{id}/relationName</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>__findById__relationName</td>
+      <td>GET</td>
+      <td>/model-name-plural/{id}/relationName</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>__count__relationName</td>
+      <td>GET</td>
+      <td>/model-name-plural/{id}/relationname</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>createChangeStream</td>
+      <td>GET</td>
+      <td>/model-name-plural/change-stream</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>createChangeStream</td>
+      <td>POST</td>
+      <td>/model-name-plural/change-stream</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>count</td>
+      <td>GET</td>
+      <td>/model-name-plural/count</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>findOne</td>
+      <td>GET</td>
+      <td>/model-name-plural/findOne</td>
+    </tr>
+    <tr>
+      <td>WRITE</td>
+      <td>upsert</td>
+      <td>PATCH</td>
+      <td>/model-name-plural</td>
+    </tr>
+    <tr>
+      <td>&nbsp;</td>
+      <td>upsert</td>
```

Comment:
correct me if I am wrong, my understanding is the `property` field should be unique, but here `upsert` maps to two endpoints: `PATCH/model-name-plural`and `PUT/model-name-plural`
I feel `replace` is the proper word for `PUT/model-name-plural`.

Reply:
In loopback 2, both `PATCH` and `PUT` map to the upsert property.  

I'm in the process of migrating our application to Loopback 3 and noticed that the behaviour of PUT has been changed.  I can update the loopback 3 documentation with the new information as soon as I've successfully completed the migration

Reply:
@frankorama sounds good! I didn't notice it's for loopback 2.